docs/contributing/code_guide.rst

```diff
@@ -279,39 +279,138 @@ to edit the pull request by changing the corresponding branch on your
 PlasmaPy fork on GitHub.  After a pull request is merged into the
```

Comment:
The coding guide is getting rather long.  I'm thinking it might be time to add a ``.. contents:` to the top of the page.  Thoughts?

Reply:
Yes!

Reply:
Addressing it in #1662 for not only the coding guide, but also the other pages in the contributor guide.